metrics/lib/common.bash

```diff
@@ -42,6 +42,47 @@ function save_results(){
 	bash $LIB_DIR/send_results.sh -n "$1" -a "$2" -r "$3" -u "$4"
 }
 
+# This function checks existence of commands.
+# They can be received standalone standalone or as an array, e.g.
+#
+# cmds=(“cmd1” “cmd2”)
+# check_cmds "${cmd[@]}"
+function check_cmds()
+{
+	req_tools="$@"
```

Comment:
Before you run check_cmds there is no global variable "req_tool". After you have run it there is. The same is true of "cmd".

In shell scripting all variables are global by default, and are dynamically scoped.  This is in contrast for example with Python, where 'req_tools=1' would create a local variable if there isn't an existing global.